File: src/main/java/net/year4000/mapnodes/api/game/modes/GameMode.java
/*
 * Copyright 2015 Year4000. All Rights Reserved.
 */

package net.year4000.mapnodes.api.game.modes;

import org.bukkit.event.Listener;

public interface GameMode extends Listener {
    public <T extends GameModeConfig> T getConfig();

    public void setConfig(GameModeConfig config);

    public GameModeConfig getRawConfig();
}

File: core/src/main/js/mapnodes/game/game_mode.js
/*
 * Copyright 2020 Year4000. All Rights Reserved.
 */
import JsonObject from './json_object.js'


/** Represents a game mode */
export default class GameMode extends JsonObject {
  /** Get the json for this kit */
  game_mode() {
    return this._json
  }
}
